main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/MessageService.cs

```diff
@@ -407,31 +407,44 @@ public static void PlaceDialog (Window childControl, Window parent)
 		/// <summary>Centers a window relative to its parent.</summary>
 		static void CenterWindow (Window childControl, Window parentControl)
 		{
+			var gtkChild = childControl?.nativeWidget as Gtk.Window;
+			var gtkParent = parentControl?.nativeWidget as Gtk.Window;
 #if MAC
-			NSWindow nsChild = childControl, nsParent = parentControl;
-			if (nsParent == null || !nsParent.IsVisible) {
-				nsChild.Center ();
-			} else {
-				int x = (int) (Math.Max (0, (nsParent.Frame.Width - nsChild.Frame.Width) / 2) + nsParent.Frame.X);
-				int y = (int) (Math.Max (0, (nsParent.Frame.Height - nsChild.Frame.Height) / 2) + nsParent.Frame.Y);
-				nsChild.SetFrame (new CoreGraphics.CGRect (x, y, nsChild.Frame.Width, nsChild.Frame.Height), true);
+			var nsChild = childControl?.nativeWidget as NSWindow;
+			var nsParent = parentControl?.nativeWidget as NSWindow;
```

Comment:
```suggestion
			var nsParent = parentControl?.nativeWidget as NSWindow;
			var nsParent = parentControl?.nativeWidget as NSWindow ?? parentControl;
```
to cover the "native dialog <-> gtk parent" case we could still convert parentControl to NSWindow implicitely